### DISCUSSION TITLE
Git segment: are bare repos supported?

### DISCUSSION THREAD

**Discussion by gitarian** [JanDeDobbeleer/oh-my-posh#2913] [Troubleshoot] [answered]
Hi,

I have a couple of bare repos serving as my outbound "gateways" to Github. When browsing those the Git segment isn't rendered indicating bare repos aren't recognized/supported yet.

Is this true or just happen to be resulting from an error on my (theme) side?

Thanks, gitarian

**Comment by JanDeDobbeleer** [comment]
This is true. Although we can add support for this. What would you expect to see in this use-case?

**Reply by gitarian** [reply]
The Git segment (--> "yes, it's a git repo") with just one dedicated icon/text, so adding just a new property `bare_repo_icon` or `no_worktree_icon` on the output/render side would already meet my needs.

Another edit: maybe an additional flag `.IsBare(Repo)` would also make some sense.

What do you/anybody think?

**Reply by JanDeDobbeleer** [reply]
Sounds like a plan. Let me see what I can do.

**Reply by JanDeDobbeleer** [reply]
I've got a working implementation, just need to add some tests.

**Reply by gitarian** [reply]
Wow, this is simply phantastic. Though I have to add, it wasn't my intention to make you code on Sunday.

Let me know when I could/should do some tests on my side.

Thanks a lot.

**Reply by JanDeDobbeleer** [reply]
@gitarian I code every day 😃 My choice, I ❤️ building tools people enjoy.

**Reply by JanDeDobbeleer** [reply]
@gitarian #2924

**Comment by gitarian** [comment, marked as answer]
Wow<sup>3</sup> :heart_eyes:

Thanks heaps! 

Already checked the displayment of bare repos, works! :heavy_check_mark:

Checking in more detail (using `.IsBare` in templates etc.) I'll do later this week.

**Reply by JanDeDobbeleer** [reply]
@gitarian you can already set `"fetch_bare_info": true` in the git segment's properties as I'm about to disable support by default as of the next version. It adds 8ms by running the check every time so I want people to consciously enable this.

**Reply by gitarian** [reply]
@JanDeDobbeleer 
does this (just) mean `.IsBare` is only evaluated when `"fetch_bare_info": true`? If yes, saving 8ms for anybody but me is the (only) way to go. On a shell every msec counts! :thumbsup:

Anyway, here's my pending feedback: `.IsBare` is working just fine. :star_struck:

During my tests I've noticed the upstream icon (if there's an upstream ref) is only shown on non-bare repos. Is this on purpose? *I know, bare repos are originally designed to provide the server-side "end-of-the-repo-chain".* I just want to understand.

Another anyway: another coffee was overdue.

**Reply by JanDeDobbeleer** [reply]
@gitarian yes to both questions. I can check if it's possible to add the origin, let me see.

**Reply by JanDeDobbeleer** [reply]
@gitarian incoming.

**Reply by gitarian** [reply]
@JanDeDobbeleer 
Thanks. Will report back after testing.

**Reply by gitarian** [reply]
> git: disable bare repo's by default ([e0b28bd](https://github.com/JanDeDobbeleer/oh-my-posh/commit/e0b28bd2c456001cd12a53fc402459e64e415bea)), closes https://github.com/JanDeDobbeleer/oh-my-posh/issues/2913
> git: fetch remote on bare repo ([040a73e](https://github.com/JanDeDobbeleer/oh-my-posh/commit/040a73e0f00bd36ae94b9ff9325992eecb3ff641))
 
2 tests passed ✔️

Thank again, Jan. :-)
